src/ui/pages/work/partusage/PartUsageDialog.js

```diff
@@ -362,10 +508,9 @@ function PartUsageDialog(props) {
         delete partUsageSubmitData.transactionInfo;
 
         // Submit the new part usage
-        WSWorkorders.createPartUsage(partUsageSubmitData)
+        await WSWorkorders.createPartUsage(partUsageSubmitData)
             .then(successHandler)
-            .catch(handleError)
-            .finally(() => setLoading(false));
+            .catch(handleError);
```

Comment:
The mix of ES6 and ES7 for asynchronous handling in the same line makes the code more confusing. If you are using async/await, I would advocate for try/catch

Reply:
True, I'll change it